repository/index.js

```diff
@@ -170,20 +169,84 @@ const getAgendaURI = async (newAgendaId) => {
   return data.results.bindings[0].agenda.value;
 };
 
+/**
+ * Deletes agendaitems for a specific agenda
+ * @name deleteAgendaitems
+ * @function
+ * @param {String} deleteAgendaURI - The URI of the agenda to delete the agendaitems from
+ */
 const deleteAgendaitems = async (deleteAgendaURI) => {
+  const agendaItemUrisQueryResult = await selectAgendaItems(deleteAgendaURI);
+  const uriList = agendaItemUrisQueryResult.results.bindings;
+  const listOfAgendaItemUris = uriList.map((uri) => { return uri.agendaitem.value});
+
+  for (const agendaItemUri of listOfAgendaItemUris) {
+    await deleteFromAgendaItems(deleteAgendaURI, agendaItemUri);
+    await deleteToAgendaItems(deleteAgendaURI, agendaItemUri);
+  }
+};
+
+/**
+ * Retrieves the agendaItem uris from an agenda
+ * @name selectAgendaItems
+ * @function
+ * @param {String} deleteAgendaURI - The URI of the agenda containing the agendaitem URIs
+ */
+const selectAgendaItems = async (deleteAgendaURI) => {
+  const query = `
+  PREFIX dct: <http://purl.org/dc/terms/>
+
+  SELECT * WHERE {
+    GRAPH <${targetGraph}> { 
+    ${sparqlEscapeUri(deleteAgendaURI)} dct:hasPart ?agendaitem .
+    }
+  }`;
+  return mu.query(query);
```

Comment:
selectAgendaItemsFromAgenda ? Deze methode klinkt herbruikbaar dus misschien de naam iets duidelijker maken.
ook: agenda**i**tems ipv agenda**I**tems. item met hoofdletter strookt niet met het model.
Er zijn nog plaatsen waar dit met hoofdletter is die ik zou willen refactoren. Liefst geen nieuwe toevoegen.